### PR TITLE
feat: add sp-repo-review

### PR DESCRIPTION
**Pull request by henryiii**
Adding sp-repo-review.

### CODE REVIEW

**Comment by danielballan** [comment]
Exciting...

**Comment by henryiii** [comment]
Just needs the testing set up, which I hope to do today.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,5 +44,11 @@ jobs:
               || '
               cookie,
               '
+            }} ${{
```
Comment (henryiii) [inline .github/workflows/ci.yml:47]:
```suggestion
            }}
            ${{
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+dynamic = ["version"]
+dependencies = [
+  "pyyaml",
+  "repo-review",
+]
+
+[project.optional-dependencies]
+cli = [
+  "click>=8",
+  "rich>=12.2",
```
Comment (henryiii) [inline pyproject.toml:38]:
```suggestion
  "repo-review[cli]",
```

**Comment by henryiii** [comment]
Now links to the guide & guide has check codes:

Screenshot 2023-06-05 at 5 59 33 PM

**Pull request merged**